pulsar-broker/src/test/java/org/apache/pulsar/client/impl/RawReaderTest.java

```diff
@@ -357,14 +359,14 @@ public void testAcknowledgeWithProperties() throws Exception {
 
         PersistentTopic topicRef = (PersistentTopic) pulsar.getBrokerService().getTopicReference(topic).get();
         ManagedLedger ledger = topicRef.getManagedLedger();
-        for (int i = 0; i < 30; i++) {
-            if (ledger.openCursor(subscription).getProperties().get("foobar").equals(0xdeadbeefdecaL)) {
-                break;
-            }
-            Thread.sleep(100);
-        }
-        Assert.assertEquals(ledger.openCursor(subscription).getProperties().get("foobar"),
-                Long.valueOf(0xdeadbeefdecaL));
+
+        Awaitility.await()
+                .atMost(5, TimeUnit.SECONDS)
+                .untilAsserted(() ->
```

Comment:
sometimes, it might be necessary to also add [`.ignoreExceptions()`](https://javadoc.io/doc/org.awaitility/awaitility/latest/org/awaitility/core/ConditionFactory.html#ignoreExceptions--) . Perhaps that would be useful if there's a chance that `ledger.openCursor(subscription).getProperties().get("foobar")` throws an exception?
By default, Awaitility will only retry by catching java.lang.AssertionErrors.

Reply:
The exceptions that I am concerned about are being handled by default.